pkg/networkutils/network.go

```diff
@@ -375,6 +376,44 @@ func (n *linuxNetwork) UpdateHostIptablesRules(vpcCIDRs []string, primaryMAC str
 	return n.updateHostIptablesRules(vpcCIDRs, primaryMAC, primaryAddr, v4Enabled, v6Enabled)
 }
 
+func (n *linuxNetwork) CleanUpStaleAWSChains(v4Enabled, v6Enabled bool) error {
+	ipProtocol := iptables.ProtocolIPv4
+	if v6Enabled {
+		ipProtocol = iptables.ProtocolIPv6
+	}
+
+	ipt, err := n.newIptables(ipProtocol)
+	if err != nil {
+		return errors.Wrap(err, "stale chain cleanup: failed to create iptables")
+	}
+
+	existingChains, err := ipt.ListChains("nat")
```

Comment:
This would get all chains and we walk thru all and skip the ones without AWS-* right..I was thinking if we can first check https://github.com/coreos/go-iptables/blob/8c6a487c56db75caf04fc043d562ea09b17315be/iptables/iptables.go#L319 if AWS-SNAT-CHAIN1 exists and then do ahead of ListChains and deleting..So we can avoid list when there is only CHAIN0..

```
iptables -t nat -S AWS-SNAT-CHAIN-1 1
-A AWS-SNAT-CHAIN-1 ! -o vlan+ -m comment --comment "AWS, SNAT" -m addrtype ! --dst-type LOCAL -j SNAT --to-source 192.168.95.7 --random-fully
```

Reply:
Ack